crates/ursa-consensus/src/server.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{AbciQueryQuery, BroadcastTxQuery};
+use multiaddr::{Multiaddr, Protocol};
+use narwhal_types::{TransactionProto, TransactionsClient};
+use tendermint_proto::abci::ResponseQuery;
+use tokio::sync::mpsc::Sender;
+use tokio::sync::oneshot::{channel as oneshot_channel, Sender as OneShotSender};
+use tracing::{debug, error};
+use warp::{Filter, Rejection, Reply};
+
+/// Simple HTTP API server which listens to messages on:
+/// * `broadcast_tx`: forwards them to Narwhal's mempool/worker socket, which will proceed to put
+/// it in the consensus process and eventually forward it to the application.
+/// * `abci_query`: forwards them over a channel to a handler (typically the application).
+pub struct AbciApi<T> {
+    mempool_address: String,
+    tx: Sender<(OneShotSender<T>, AbciQueryQuery)>,
+}
+
+impl<T: Send + Sync + std::fmt::Debug> AbciApi<T> {
+    pub async fn new(
+        mempool_address: Multiaddr,
+        tx: Sender<(OneShotSender<T>, AbciQueryQuery)>,
+    ) -> Self {
+        let mempool_port = mempool_address
+            .iter()
+            .find_map(|proto| match proto {
+                Protocol::Tcp(port) => Some(port),
+                _ => None,
+            })
+            .expect("Expected tcp url for worker mempool");
+
+        let mempool_address = format!("http://0.0.0.0:{}", mempool_port);
```

Comment:
Is this a re-instantiation of the variable `mempool_address`? If so, could we rename it for readability? its based into the new function as type  `Multiaddr` and stored as type `String`